pkg/ring/lifecycler.go

```diff
@@ -456,6 +462,21 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 
 		ingesterDesc, ok := ringDesc.Ingesters[i.ID]
 		if !ok {
+			var tokens Tokens
+			// We load the tokens from the file only if it does not exist in the ring yet.
+			err := tokens.LoadFromFile(i.cfg.TokensFilePath)
```

Comment:
Longer the CAS operation takes, the higher chance of it failing is, and then entire load needs to be repeated. Perhaps this decision can be moved outside of CAS, so that Load is only performed once?

Reply:
Right. Didn't notice that part. Thanks!